compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -142,73 +139,51 @@ impl<'a> ModCollector<'a> {
         impls: Vec<NoirTraitImpl>,
         krate: CrateId,
     ) -> Vec<(CompilationError, fm::FileId)> {
-        let module_id = ModuleId { krate, local_id: self.module_id };
-        let mut diagnostics: Vec<(CompilationError, fm::FileId)> = vec![];
-
         for trait_impl in impls {
-            let trait_name = &trait_impl.trait_name;
-            let module = &self.def_collector.def_map.modules[self.module_id.0];
-
-            match self.find_trait_or_emit_error(module, trait_name) {
-                Ok(trait_id) => {
-                    let collected_trait =
-                        self.def_collector.collected_traits.get(&trait_id).cloned().unwrap();
-                    let unresolved_functions = self.collect_trait_implementations(
-                        context,
-                        &trait_impl,
-                        &collected_trait.trait_def,
-                        krate,
-                        &mut diagnostics,
-                    );
-
-                    for (_, func_id, noir_function) in &unresolved_functions.functions {
-                        let function = &noir_function.def;
-                        context.def_interner.push_function(*func_id, function, module_id);
-                    }
-
-                    let unresolved_trait_impl = UnresolvedTraitImpl {
-                        file_id: self.file_id,
-                        module_id: self.module_id,
-                        the_trait: collected_trait,
-                        methods: unresolved_functions,
-                        trait_impl_ident: trait_impl.trait_name.clone(),
-                    };
-
-                    let key = (trait_impl.object_type, self.module_id, trait_id);
-                    self.def_collector.collected_traits_impls.insert(key, unresolved_trait_impl);
-                }
-                Err((err, file_id)) => diagnostics.push((err.into(), file_id)),
+            let trait_name = trait_impl.trait_name.clone();
+
+            let unresolved_functions =
+                self.collect_trait_impl_function_overrides(context, &trait_impl, krate);
+
+            for (_, func_id, noir_function) in &unresolved_functions.functions {
+                let name = noir_function.name().to_owned();
+
+                let modifiers = FunctionModifiers {
+                    name: name.clone(),
+                    visibility: crate::Visibility::Public,
+                    attributes: Attributes::empty(),
+                    is_unconstrained: false,
+                    contract_function_type: None,
+                    is_internal: None,
+                };
+                context.def_interner.push_function_definition(
```

Comment:
You don't need to manually create the FunctionModifiers here, since you have the noir_function you can use it's def field and call `context.def_interner.push_function` instead

Reply:
Fixed [dcffa07](https://github.com/noir-lang/noir/pull/2844/commits/dcffa0711a98c51613fad2c5dbeba74a50ac6ae2)